### PR TITLE
Fwup error fix

### PR DESCRIPTION
**Pull request by ConnorRigby**
Don't actually merge this. 

# What i did
Wrote a tiny script that wraps a command allowing it to successfully complete before closing file descriptors.

# why
There seems to be a race condition in Erlangs :port implementation where the file descriptor closing message gets dispatched before the final IO data message. This causes the Port to be closed before receiving the error message from FWUP.

### CODE REVIEW

File: priv/wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+$@
+EXIT=$?
+# We don't have sleep..
+dd if=/dev/zero of=/dev/null bs=100M count=10
```
Comment (fhunleth) [inline priv/wrapper.sh]: omg

**Comment by ConnorRigby** [comment]
Please don't judge me

**Comment by fhunleth** [comment]
Not judging. I think this is great. You dug into this enough to get me ideas on how to fix this more cleanly and also to understand what Erlang is doing behind the scenes. I think that I have more respect for Erlang now but I'm going to double check my hunch tomorrow when my brain is fresh.

Creative use of `dd` to simulate a sleep... It made me remember than I hadn't enabled it in Busybox.

**Comment by ConnorRigby** [comment]
It worked _pretty_ reliably without the simulated sleep. But with about 1 or 2 seconds it's battle ready. :|

**Comment by fhunleth** [comment]
Thanks! I added a handshake with `fwup` that fixes the fundamental problem and the change is in #12. Your debug was critical for me to see this.

**Pull request closed without merging**